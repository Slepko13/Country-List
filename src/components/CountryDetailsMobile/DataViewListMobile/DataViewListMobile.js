import React from 'react';
import './DataViewListMobile.scss';

import PropTypes from "prop-types";


const DataViewListMobile = (props) => {

    let {title, data, position, addictions, background, picture, vectorHeight} = props;

    if (!data) return <div
        className={position ?
            `DataViewListMobile ${position} ` :
            "DataViewListMobile"}
        data-testid="position"
    >
        <div className="absent">Data is absent today</div>
    </div>;

    return (
        <div className={position ? `DataViewListMobile  ${position} ` : "DataViewListMobile"}>
            <img className="ellipse" src={picture} alt='ellipse'/>
            <div className="innerVector" style={{height: vectorHeight}}/>
            <div className="info__title">{title}</div>
            <div className="info__content">{
                data.length ?
                    data.filter(item => item.name !== "null")
                        .map(item =>
                            <div
                                className="item"
                                key={item.name}
                                style={{backgroundColor: background}}
                            >{addictions}
                                {item.symbol &&
                                <span className="item__symbol">{
                                    item.symbol === "null" ? "n/a" : `"${item.symbol}"`
                                }</span>
                                }

                                <span> {item.name.replace("UTC", "GMT")}</span>

                            </div>)
                    :
                    <div className="item">n/a</div>
            }
            </div>
            {/*{data.length > 3 ? <div style={{height: "20px"}}/> : null}*/}
        </div>
    )
}

export default DataViewListMobile;

DataViewListMobile.propTypes = {
    title: PropTypes.string,
    data: PropTypes.array,
    position: PropTypes.string,
    addictions: PropTypes.string,
    background: PropTypes.string,
    picture: PropTypes.string,
    vectorHeight: PropTypes.string
}