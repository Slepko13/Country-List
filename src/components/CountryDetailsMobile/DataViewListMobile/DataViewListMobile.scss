.DataViewListMobile {
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin: 0 0 25px 90px;

  .ellipse {
    position: absolute;
    top: 12px;
    right: 100%;
    margin: 0 25px 0 0;
  }

  .innerVector {
    position: absolute;
    width: 12px;
    height: 98%;
    background: url("../../../assets/images/Vector 10.png") center top;
    top: 38px;
    right: 100%;
    margin: 1px 25px 0 0;
    transform: translateX(-50%);
  }

  .info__title {
    font-family: "Roboto", sans-serif;
    font-style: normal;
    font-weight: 300;
    font-size: 14px;
    line-height: 16px;
    text-align: center;
  }

  .info__content {
    position: relative;
    font-family: "Roboto", sans-serif;
    font-style: normal;
    font-weight: bold;
    text-align: center;
    color: #313131;
    display: flex;
    justify-content: flex-start;
    flex-wrap: wrap;

    .item {
      text-align: left;
      font-size: 20px;
      line-height: 23px;
      text-overflow: ellipsis;
      overflow: hidden;
      margin: 5px 10px 5px 0;
      border-radius: 18px;
      padding: 3px 10px;

      &:hover {
        overflow: visible;
        max-width: 800px;
        background-color: #ffffff;
      }

      .item__symbol {
        margin-right: 10px;
      }
    }
  }
}