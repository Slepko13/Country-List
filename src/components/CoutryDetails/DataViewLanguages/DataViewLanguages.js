import React from 'react';
import PropTypes from "prop-types";
import './DataViewLanguages.scss';
import SeeMore from "../SeeMore/SeeMore";

const DataViewLanguages = (props) => {

    let { title, data, position } = props;
    let myData, restData;
    if (!data) return <div className={position ? `DataViewLanguages ${position}` : "DataViewLanguages"} data-testid="position">
        <div className="absent">Data is absent today</div>
    </div>;

    if (data.length <= 3 && data.length >= 0) {
        myData = [...data];
    } else {
        myData = data.slice(0, 3);
        restData = data.slice(3);
    }
    return (
        <div className={position ? `DataViewLanguages + ${position} ` : "DataViewLanguages"}
        >
            <div className="info__title">{title}</div>
            <div className="info__content">{
                myData.length ?
                    myData.map(item =>
                        <div
                            className="item"
                            key={item.name}
                        >{item.name}</div>) :
                    <div className="item">n/a</div>
            }
                {restData ? <div className="wrapper">
                    <SeeMore
                        data={restData}
                        bottom="20%"
                        top=""
                        left=""
                        right=""
                    /></div> : null}

            </div>
        </div>
    )
}

export default DataViewLanguages;

DataViewLanguages.propTypes = {
    title: PropTypes.string,
    data: PropTypes.array,
    position: PropTypes.string
}