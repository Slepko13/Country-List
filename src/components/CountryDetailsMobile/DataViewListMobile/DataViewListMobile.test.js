import React from "react";
import DataViewListMobile from "./DataViewListMobile";
import {render} from "@testing-library/react";
import DataViewLanguages from "../DataViewLanguagesMobile/DataViewLanguagesMobile";


describe("DataViewListMobile component", () => {
    it("should renders with  additional classname depending on props ", () => {
        const { container, getByTestId } = render(
            <DataViewListMobile position="left" />
        );
        expect(getByTestId('position')).toHaveClass('left');
    });

    it("should render  without additional classname depending on props ", () => {
        const { container, getByTestId } = render(
            <DataViewListMobile position={null} />
        );
        expect(container.firstChild.classList.contains('left')).toBe(false)
    });
    describe("symbol is absent", ()=>{
        const props = {
            data : [
                {name :"english" },
                {name :"french" },
            ],
            title :"Languages",
            position: "left"
        }
        const component = shallow(<DataViewListMobile {...props}/>);
        it("should renders correct symbol text", () => {
            expect(component.find('.info__content').childAt(0).find('.item__symbol').text()).toEqual('');
        });
    });
    describe("symbol is string 'null' ", ()=>{
        const props = {
            data : [
                {name :"english", symbol: "null" },
                {name :"french",  symbol: "null" },
            ],
            title :"Languages",
            position: "left"
        }
        const component = shallow(<DataViewListMobile {...props}/>);
        it("should renders correct symbol text", () => {
            expect(component.find('.info__content').childAt(0).find('.item__symbol').text()).toEqual('n/a');
        });
    });
    describe("symbol renders properly ", ()=>{
        const props = {
            data : [
                {name :"english", symbol: "$" },
                {name :"french",  symbol: "â‚´" },
            ],
            title :"Languages",
            position: "left"
        }
        const component = shallow(<DataViewListMobile {...props}/>);
        it("should renders correct symbol text", () => {
            expect(component.find('.info__content').childAt(0).find('.item__symbol').text()).toEqual('"$"');
        });
    });

    describe("position is absent", ()=>{
        const props = {
            data : [
                {name :"english" },
                {name :"french" },
            ],
            title :"Languages",
            // position: "left"
        }
        const component = shallow(<DataViewListMobile {...props}/>);
        it("should renders correct className", () => {
         expect(component.find('.DataViewListMobile')).toHaveLength(1);
        });
    });
    describe("data is absent", ()=>{
        const props = {
            title :"Languages",
            position: "left"
        }
        const component = shallow(<DataViewListMobile {...props}/>);
        it("should renders 'Data is absent today ", () => {
            expect(component.find('.absent').text()).toEqual('Data is absent today');
        });
    });
    describe("data is empty array", ()=>{
        const props = {
            data :[],
            title :"Languages",
            position: "left"
        }
        const component = shallow(<DataViewListMobile {...props}/>);
        it("should renders 'n/a ", () => {
            expect(component.find('.item').text()).toEqual('n/a');
        });
    });
    describe("data length <= 3", ()=>{
        const props = {
            data : [
                {name :"english" },
                {name :"french" },
            ],
            title :"Languages",
            position: "left"
        }
        const component = shallow(<DataViewListMobile {...props}/>);

        it("should renders 2 items ", () => {
            expect(component.find('div.item')).toHaveLength(2);
        });
        it("should renders properly ", () => {
            expect(component).toMatchSnapshot();
        });
    });
    describe(" data length > 3",() => {
        const props = {
            data : [
                {name :"english" , symnol: "e"},
                {name :"french" , symnol: "f"},
                {name :"german", symnol: "g" },
                {name :"polish", symnol: "p" }
            ],
            title :"Languages",
            position: "left"
        }
        const component = shallow(<DataViewListMobile {...props}/>);

        it("should renders properly ", () => {
            expect(component).toMatchSnapshot();
        });

        it("should renders 3 items and SeeMore ", () => {
            expect(component.find('div.item')).toHaveLength(3);
            expect(component.find('SeeMore')).toHaveLength(1);
        });
    });
});

