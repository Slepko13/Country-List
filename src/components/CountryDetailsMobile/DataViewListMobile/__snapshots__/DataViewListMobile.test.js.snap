// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataViewListMobile component  data length > 3 should renders properly  1`] = `
<div
  className="DataViewListMobile  left "
>
  <img
    alt="ellipse"
    className="ellipse"
  />
  <div
    className="innerVector"
    style={
      Object {
        "height": undefined,
      }
    }
  />
  <div
    className="info__title"
  >
    Languages
  </div>
  <div
    className="info__content"
  >
    <div
      className="item"
      key="english"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      <span>
         
        english
      </span>
    </div>
    <div
      className="item"
      key="french"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      <span>
         
        french
      </span>
    </div>
    <div
      className="item"
      key="german"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      <span>
         
        german
      </span>
    </div>
    <div
      className="item"
      key="polish"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      <span>
         
        polish
      </span>
    </div>
  </div>
</div>
`;

exports[`DataViewListMobile component data length <= 3 should renders properly  1`] = `
<div
  className="DataViewListMobile  left "
>
  <img
    alt="ellipse"
    className="ellipse"
  />
  <div
    className="innerVector"
    style={
      Object {
        "height": undefined,
      }
    }
  />
  <div
    className="info__title"
  >
    Languages
  </div>
  <div
    className="info__content"
  >
    <div
      className="item"
      key="english"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      <span>
         
        english
      </span>
    </div>
    <div
      className="item"
      key="french"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      <span>
         
        french
      </span>
    </div>
  </div>
</div>
`;
